manifests/tftp/efi.pp

```diff
@@ -0,0 +1,66 @@
+# EFI support for TFTP
+class foreman_proxy::tftp::efi (
+  Enum['package', 'redhat_grub', 'redhat_grub_old', 'debian_grub', 'none'] $installation_type = $::foreman_proxy::tftp::efi::params::installation_type,
+  Array[String] $grub_packages = $::foreman_proxy::tftp::efi::params::grub_packages,
+  String $grub_modules = $::foreman_proxy::tftp::efi::params::grub_modules,
+  Stdlib::Absolutepath $root = $::foreman_proxy::tftp::root,
+) inherits foreman_proxy::tftp::efi::params {
+  ensure_packages($grub_packages, { ensure => 'present', })
+
+  case $installation_type {
+    'package': {
+      $family = downcase($::osfamily)
+      package { "foreman-bootloaders-${family}-tftpboot":
+        ensure => installed,
+      }
+    }
+    'redhat_grub': {
+      $grub_efi_path = $::operatingsystem ? {
+        /Fedora|CentOS/ => downcase($::operatingsystem),
+        default         => 'redhat',
+      }
+
+      file { "${root}/grub2/grubx64.efi":
+        ensure => file,
+        source => "/boot/efi/EFI/${grub_efi_path}/grubx64.efi",
+      }
+
+      file { "${root}/grub2/shim.efi":
+        ensure => file,
+        source => "/boot/efi/EFI/${grub_efi_path}/shim.efi",
+      }
+    }
+    'redhat_grub_old': {
+      file {"${root}/grub/grubx64.efi":
+        ensure => file,
+        owner  => 'root',
+        mode   => '0644',
+        source => '/boot/efi/EFI/redhat/grub.efi',
+      }
+
+      file {"${root}/grub/shim.efi":
+        ensure => 'link',
+        target => 'grubx64.efi',
+      }
+    }
+    'debian_grub': {
```

Comment:
Although we do not provide debian package yet, tarballs are downloadable from http://downloads.theforeman.org/foreman-bootloaders/ for debian as well. Might be also an option for debian users.

Reply:
I do wonder why there's the OS name in the package name. Why can't it just be yum/apt install foreman-bootloaders?

Reply:
Distributions carries a lot (hundreds) of patches in grub2, redhat distribution is very different from Debian/Ubuntu. These builds were taken from redhat, thus the name. You can install debian versions next to those when we make them available.